docker/docker_client.go

```diff
@@ -129,12 +138,29 @@ func newTransport() *http.Transport {
 	return tr
 }
 
+// dockerCertDir returns a path to a directory to be consumed by setupCertificates() depending on ctx and hostPort, or "" if no directory should be used.
+func dockerCertDir(ctx *types.SystemContext, hostPort string) string {
+	if ctx != nil && ctx.DockerCertPath != "" {
+		return ctx.DockerCertPath
+	}
+	var hostCertDir string
+	if ctx != nil && ctx.DockerPerHostCertDirPath != "" {
+		hostCertDir = ctx.DockerPerHostCertDirPath
+	} else if ctx != nil && ctx.RootForImplicitAbsolutePaths != "" {
+		hostCertDir = filepath.Join(ctx.RootForImplicitAbsolutePaths, systemPerHostCertDirPath)
+	} else {
+		hostCertDir = systemPerHostCertDirPath
```

Comment:
The quoted line (`hostCertDir = systemPerHostCertDirPath`) is actually expected to be the common case. But the debug log _should_ record which path was tried, thanks for the reminder.